### ISSUE TITLE
Why I got loss value nan in working with citeseer data

### ISSUE THREAD

**Issue by gaudelbijay** [danielegrattarola/keras-gat#28]
Epoch 1/100
3327/3327 - 2s - loss: nan - categorical_crossentropy: nan - acc: 0.1667 - val_loss: nan - val_categorical_crossentropy: nan - val_acc: 0.0580
Epoch 2/100
3327/3327 - 2s - loss: nan - categorical_crossentropy: nan - acc: 0.1667 - val_loss: nan - val_categorical_crossentropy: nan - val_acc: 0.0580

**Comment by danielegrattarola** [comment]
Hi,

can you provide more details on this? Did you only change the dataset? What versions of TF and Keras are you running?

Cheers. 
Daniele

**Comment by gaudelbijay** [comment]
I am running tensorflow 2.2.0

**Comment by danielegrattarola** [comment]
I can run the code on TF 2.2 and Keras 2.4.3 and it trains as expected with no NaNs.
Are you running on GPU with some particular setup (e.g. restricted precision)? Does it work if you train on CPU only? 

FYI, expect the code not to work on Pubmed because this implementation of GAT is very inefficient. 
You can have a look at Spektral, my library for GNNs in tf.keras. We have an efficient and up-to-date implementation of GAT (see [docs](https://graphneural.network/layers/convolution/#graphattention)) that will work on Pubmed as well.

**Comment by gaudelbijay** [comment]
Why this gives very bad result in the inductive settings for the citeseer dataset. Is there any specific reason?

**Comment by danielegrattarola** [comment]
Can you be a bit more specific? Citeseer is a transductive task and the code should reproduce the results from the paper as-is, I'm not sure I understand that problem that you are encountering. 

Anyway, were you able to solve the original issue? 

Cheers

**Comment by gaudelbijay** [comment]
Yes I solve the problem of loss value. But when I trained Citeseer dataset in inductive setting according to the guideline given in the paper for inductive setting. I didn't get good result as in transductive setting.

**Comment by gaudelbijay** [comment]
Can't we trained Citeseer in inductive settings? Like Citatio  data in GraphSAGE paper?
